src/futhark.nim

```diff
@@ -33,6 +33,47 @@ const
   preAnsiFuncDecl = defined(preAnsiFuncDecl)
   echoForwards = defined(echoForwards)
 
+when not declared(buildOS):
+  const buildOS {.magic: "BuildOS".}: string = hostOS
+const windowsHost = buildOS == "windows"
+
+func hostQuoteShell(s: string): string =
+  ## Quote ``s``, so it can be safely passed to shell.
+  if windowsHost:
+    result = quoteShellWindows(s)
+  else:
+    result = quoteShellPosix(s)
+
+proc hostQuoteShellCommand(args: openArray[string]): string =
+  ## Concatenates and quotes shell arguments `args`.
+  result = ""
+  for i in 0..<args.len:
+    if i > 0: result.add " "
+    result.add hostQuoteShell(args[i])
+
+proc hostIsAbsolute(path: string): bool =
+  ## Checks whether a given `path` is absolute.
+
+  if len(path) == 0: return false
+
+  if windowsHost:
```

Comment:
Should this be `when not defined(windows) and windowsHost`? So that if you're compiling on Windows for Windows you will still use the standard library `isAbsolute`?